crates/sui-framework/packages/sui-system/tests/validator_set_tests.move

```diff
@@ -313,90 +300,171 @@ module sui_system::validator_set_tests {
     }
 
     #[test]
-    fun test_low_stake_departure() {
+    #[expected_failure(abort_code = validator_set::ENotAValidator)]
+    fun test_request_add_then_pull_stake() {
```

Comment:
@emmazzz I added a test for this (didn't see one previously), and wasn't sure what we would expect the outcome to be. A failure in `request_withdraw_stake` (what happens) seems like a reasonable answer, but was wondering if we should use a more descriptive error code than `ENotAValidator`. The specific problem here is (I think) that the validator is pending addition, and the stake of a pending validator cannot be withdrawn until the next epoch (when the validator will have been added)